src/main/tetris_board.cpp

```diff
@@ -315,11 +287,8 @@ void TetrisBoard::update_blocks( std::vector<int> full_rows )
       {
         if( block.get_row() < row )
         {
-          while( !has_landed( block ) )
-          {
-            occupied_spaces.at( block.get_row() ).at( block.get_column() ) = false;
-            block.fall();  
-          }
+          occupied_spaces.at( block.get_row() ).at( block.get_column() ) = false;
```

Comment:
This makes the game more Tetris-like. With the while statement, a block will fall until impact and possibly knock out another row. Without the while, blocks only fall the number of rows that were removed.